test/lib/commands/install.js

```diff
@@ -400,3 +408,148 @@ t.test('should show install keeps dirty --workspace flag', async t => {
   assert.packageDirty('node_modules/abbrev@1.1.0')
   assert.packageInstalled('node_modules/lodash@1.1.1')
 })
+
+t.test('devEngines', async t => {
```

Comment:
The PR description mentions a key difference between the `engines` and `devEngines` checks: Only the `engines` check is applied to packages being installed.

It would be good to have some tests that reiterate that behavior and prevent regressions. If the current package.json has `devEngines` that require `node` and `npm` and the package being installed (e.g. `abbrev`) has a `devEngines` that requires `nondescript`: no problem! Best to test this for both `dependency` and `devDependency` installs.

Almost the inverse of this test:
https://github.com/npm/cli/blob/fefd509992a05c2dfddbe7bc46931c42f1da69d7/test/lib/commands/install.js#L157-L188